tests/components/smartthings/test_cover.py

```diff
@@ -140,6 +142,37 @@ async def test_set_cover_position(hass: HomeAssistant, device_factory) -> None:
     assert device._api.post_device_command.call_count == 1  # type: ignore
 
 
+async def test_set_cover_position(hass: HomeAssistant, device_factory) -> None:
+    """Test the cover sets to the specific position."""
+    # Arrange
+    device = device_factory(
+        "Shade",
+        [Capability.window_shade, Capability.battery, Capability.window_shade_level],
+        {
+            Attribute.window_shade: "opening",
+            Attribute.battery: 95,
+            Attribute.shade_level: 10,
+        },
+    )
+    await setup_platform(hass, COVER_DOMAIN, devices=[device])
+    # Act
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_SET_COVER_POSITION,
+        {ATTR_POSITION: 50, "entity_id": "all"},
+        blocking=True,
+    )
+
+    state = hass.states.get("cover.shade")
+    # Result of call does not update state
+    assert state.state == STATE_OPENING
+    assert state.attributes[ATTR_BATTERY_LEVEL] == 95
+    assert state.attributes[ATTR_CURRENT_POSITION] == 10
+    # Ensure API called
+
+    assert device._api.post_device_command.call_count == 1  # type: ignore
```

Comment:
I realize this test is following the same pattern as the existing tests, but wouldn't it make sense to check the parameters of `post_device_command` also to check the correct command and correct parameters are used?

This can be done in a separate PR.

Reply:
Good call. I don't really like the leaky abstraction of digging into the guts of the library to mock, so will explore mocking the API actually used by HA in the future.